plat/drivers/virtio/virtio_pci.c

```diff
@@ -273,10 +273,12 @@ static int vpci_legacy_pci_config_get(struct virtio_dev *vdev, __u16 offset,
 				VIRTIO_PCI_CONFIG_OFF + offset, buf, len,
 				type_len);
 	} else {
+		__u32 buf_len_bytes = len * type_len;
+
 		rc = virtio_cread_bytes_many(
 				(void *) (unsigned long)vpdev->pci_base_addr,
-				VIRTIO_PCI_CONFIG_OFF + offset,	buf, len);
-		if (rc != (int)len)
+				VIRTIO_PCI_CONFIG_OFF + offset,	buf, buf_len_bytes);
+		if (rc != (int)(buf_len_bytes))
```

Comment:
The parenthesis around `buf_len_bytes` here are not needed. And while you're there, maybe wrap the check around `unlikely` to help with performance a bit.